shipsim_sensor_module/simulated_mmg_sensor_node.py

```diff
@@ -0,0 +1,55 @@
+#! /usr/bin/python3
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+import rclpy
+from rclpy.node import Node
+
+from geometry_msgs.msg import Twist
+
+class SensorNode(Node):
+    """SensorNode."""
+
+    cmd_vel_Twist = Twist()
+    cmd_vel_Twist2 = Twist()
+
+    def __init__(self, publish_address="/ship1sim/sensor", timer_period=0.1):
+        """init."""
+        super().__init__("simulated_sensor")
+        self.delta_time = timer_period
+        self.pub_sensor = self.create_publisher(Twist, publish_address, 1)
+        self.subscription = self.create_subscription(Twist, "/ship1/cmd_vel", self.listener_callback, 1)
+        self.timer = self.create_timer(timer_period, self.sender_callback)
+
+    def sender_callback(self):
+        """sender_callback."""
+
+        σ_true = 0.1 #0.01 # [rad/s] 正規分布ノイズの標準偏差
+
+        self.cmd_vel_Twist2.linear.x = self.cmd_vel_Twist.linear.x
+        self.cmd_vel_Twist2.linear.y = self.cmd_vel_Twist.linear.y
+        self.cmd_vel_Twist2.angular.z = self.cmd_vel_Twist.angular.z + np.random.normal(0, σ_true) #回頭角速度にセンサ誤差を付加

```

Comment:
Please add independent noise of each param as follows.

```py
σ_u = 0.1
σ_u = 0.1
σ_u = 0.1
self.cmd_vel_Twist2.linear.x = self.cmd_vel_Twist.linear.x + np.random.normal(0, σ_u)
self.cmd_vel_Twist2.linear.y = self.cmd_vel_Twist.linear.y + np.random.normal(0, σ_v)
self.cmd_vel_Twist2.angular.z = self.cmd_vel_Twist.angular.z + np.random.normal(0, σ_r)
```

Reply:
In addition, `σ_u`, `σ_v`, `σ_r` should be defined as global variables on around line 16.